frame/contracts/src/storage/meter.rs

```diff
@@ -255,16 +273,24 @@ impl<T, E, S> RawMeter<T, E, S>
 where
 	T: Config,
 	E: Ext<T>,
-	S: State,
+	S: State + Default + Debug,
 {
-	/// Create a new child that has its `limit` set to whatever is remaining of it.
+	/// Create a new child that has its `limit`.
+	/// Passing `0` as the limit is interpreted as to take whatever is remaining from its parent.
```

Comment:
Quite an opposite, in general passing a number is better. Passing an Option instead of a number to  call() extrinsic is just a tradeoff for keeping it work with existing UI

Reply:
I also don't see that using `0` for something else than `0` is very straight forward

Reply:
It is done here just like it is in the [gas meter](https://github.com/paritytech/substrate/blob/db6ebf564bcdfdfaf5fd026bb321de6f7d7e6fc0/frame/contracts/src/gas.rs#L105): 

https://github.com/paritytech/substrate/blob/db6ebf564bcdfdfaf5fd026bb321de6f7d7e6fc0/frame/contracts/src/gas.rs#L100-L105